### ISSUE TITLE
Model performance degradation when switch from `data` to `voting` in distributed learning algorithm

### ISSUE THREAD

**Issue by falcon8241** [microsoft/LightGBM#5769]
Not sure if this is a bug or question: I fit two DaskLGBMRegressor models using the same data and same hyper-parameters. The only difference is to switch the `tree_learner='data'` to `tree_learner='voting'`. However this causes a significant performance degradation: the R^2 drops from .931 to -5.28e+69. I am wondering what is the best course of actions to understand the reason behind this degradation. Since the data cannot be shared, what debugging can I do to dig deeper into this?

**Comment by guolinke** [comment]
voting based parallel is better to used with large #feature and #row data. When #row is small, the accuracy may is not good. when #feature is small, its speed-up is will be marginal.

**Comment by falcon8241** [comment]
@guolinke The data has over 400 features and about 2.2M rows. That's why I was confused as the sample size is not really small.

**Comment by guolinke** [comment]
Did you fully shuffle the rows before row partitions? How many nodes you used?

**Comment by falcon8241** [comment]
I did not shuffle the data. But why does data parallel seem immune to this issue? In both cases, I am using a dask cluster with 25 workers and 4 cores per worker.On Mar 7, 2023, at 7:52 AM, Guolin Ke ***@***.***> wrote:﻿
Did you fully shuffle the rows before row partitions? How many nodes you used?

**Comment by guolinke** [comment]
Voting parallel will leverage local data partition to find split candidates, and the a voting is conducted on the local top splits. Therefore, if the distribution of local data is very different with global data, the local top splits could be wrong, subsequently causing the bad performance.
The data parallel didn't have voting, so it is lossless ideally.

**Comment by falcon8241** [comment]
I reran the experiment with shuffled data, there is still performance degradation R^2 is -5.59e+30.On Mar 7, 2023, at 11:10 AM, Guolin Ke ***@***.***> wrote:﻿
Voting parallel will leverage local data partition to find split candidates, and the a voting is conducted on the local top splits. Therefore, if the distribution of local data is very different with global data, the local top splits could be wrong, subsequently causing the bad performance.
The data parallel didn't have voting, so it is lossless ideally.
